Bottom: 3f27ccad23899fdb7fcaad3e73e0de8d9afb3a52
Top:    393e180c0cd927513742ea2abc28d3837733d38a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-25 01:24:23 +0000

Fix undo for netlist selection

Oops.. broke this before. Should test more before committing!


---

diff --git a/src/find.c b/src/find.c
index f5d91d0..82b3aa8 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3405,6 +3405,7 @@ void
   User = undo;
   DumpList ();
   ListStart (type, ptr1, ptr2, ptr3);
+  printf ("RatFindHook, user=%s\n", User ? "true" : "false");
   DoIt (AndRats, false);
   User = false;
 }
@@ -3570,6 +3571,8 @@ ResetConnections (bool AndDraw)
   change = ResetFoundPinsViasAndPads  (AndDraw) || change;
   change = ResetFoundLinesAndPolygons (AndDraw) || change;
 
+  printf ("flag = %i\n", TheFlag);
+
   return change;
 }
 
diff --git a/src/select.c b/src/select.c
index 27a590b..4a5b415 100644
--- a/src/select.c
+++ b/src/select.c
@@ -1015,7 +1015,7 @@ SelectObjectByName (int Type, char *Pattern, bool Flag)
   if (Type & NET_TYPE)
     {
       InitConnectionLookup ();
-      changed = ResetConnections (true) || changed;
+      if (ResetConnections (true)) changed = true;
 
       MENU_LOOP (&PCB->NetlistLib);
       {
@@ -1034,8 +1034,8 @@ SelectObjectByName (int Type, char *Pattern, bool Flag)
       }
       END_LOOP;
 
-      changed = SelectConnection (Flag) || changed;
-      changed = ResetConnections (false) || changed;
+      if (SelectConnection (Flag)) changed = true;
+      if (ResetConnections (false)) changed = true;
       FreeConnectionLookupMemory ();
     }
